sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/events.scala

```diff
@@ -110,7 +120,31 @@ case class RenameTableEvent(
   extends TableEvent
 
 /**
- * Event fired when a function is created, dropped or renamed.
+ * Enumeration to indicate which part of table is altered. If a plain alterTable API is called, then
+ * type will generally be Table.
+ */
+object AlterTableKind extends Enumeration {
+  val Table, DataSchema, Stats = Value
```

Comment:
shall we just use string?

Reply:
I'm OK to use String, but I'd prefer strong type to avoid nasty issues.

Reply:
String is better for backward compatibility, but easier to make mistake. I don't have a strong preference, cc @hvanhovell @gatorsmile